src/NzbDrone.Common/UniqueIdentifier/UniqueIdentifierGenerator.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using NzbDrone.Common.Cache;
+
+namespace NzbDrone.Common.UniqueIdentifier
+{
+    public interface IUniqueIdentifierGenerator
+    {
+        int Get(string key);
+    }
+
+    public class UniqueIdentifierGenerator : IUniqueIdentifierGenerator
+    {
+        private static int _lastIdentifier;
+
+        private ICached<int> _cache;
+
+        public UniqueIdentifierGenerator(ICacheManager cacheManager)
+        {
+            _cache = cacheManager.GetCache<int>(typeof(UniqueIdentifierGenerator));
+        }
+
+        public int Get(string key)
+        {
+            return _cache.Get(key, GetNextIdentifier, TimeSpan.FromHours(24));
```

Comment:
An simple Math.Abs on the original code would've done that, but that doesn't prevent collisions.
The point of this PR was to end that once and for all. At least, not until 2 billion unique items have been shown on the UI.
I just wanted something that was guaranteed unique during the session.